### ISSUE TITLE
Allow setting via meta http-equiv

### ISSUE THREAD

**Issue by AshleyScirra** [WICG/feature-policy#25]
The current spec prevents setting a feature-policy based on the meta tag.

> The feature policy needs to be set before the response is processed and before any subresource requests are initiated. Once active the policy also needs to apply for the lifetime of the document and is not allowed to be modified at runtime. Due to these requirements, meta element cannot be used to set the policy.

I think this is unfortunate. There are some feature policies that probably deserve to be used in every new web app by default. For example I think there is a good argument that `docwrite`, `sync-script` and `sync-xhr` should be disabled for all new projects; these features exist mainly for backwards compatibility and are best avoided. In issue 24 I also suggest further potentially universal policies like `force-strict-mode`, `no-sync-storage` and `no-prefixed-features`.

While all professional web developers will have control over their HTTP headers, it's worth remembering the web has a broader audience than professionals. Beginners just starting out, developers working with third-party hosting servers, or even experienced developers just sharing a demo on something like a public Dropbox folder can not enforce any feature policy, because they don't know how or aren't allowed to change HTTP headers.

I think an even better argument is that it prevents templates from setting common-sense feature policies. Templates like the Twitter bootstrap, HTML5 boilerplate and others only provide the HTML files, but have no direct control over the HTTP headers. Also beginner's tutorials that want to recommend this as a kind of "use strict" mode for HTML have to delve in to HTTP configuration, instead of just saying "paste this at the top of every HTML file to turn off bad features". So while a common-sense universal feature policy might disable `docwrite`, `sync-script`, `sync-xhr` and possibly others, this cannot easily propagate through templates or guides. This means more developers end up working with these features enabled (possibly accidentally), reducing the effectiveness of their web apps, and ultimately making it harder for browsers to remove these features.

There are technical reasons to disallow setting the meta tag or dynamically changing it, but I think it would still be acceptable to allow the tag but have some restrictions on it:

 - must be specified before any script or sub-resource references or additional connections (so it can affect them)
 - must occur within the first ~4k of HTML content (like the charset) so the search for the tag is bounded
 - cannot be changed dynamically (meta viewport was like this for a long time in some browsers)

In other words, it has to be one of the first tags in the document (probably next after charset) but this still allows for the "paste-in good practices" uses outlined above.

WDYT?

**Comment by igrigorik** [comment]
@AshleyScirra could we merge this with https://github.com/igrigorik/feature-policy/issues/25?

**Comment by AshleyScirra** [comment]
Do you mean #15? I didn't see that, yeah, looks like the same issue sorry!

**Comment by igrigorik** [comment]
Sorry, yes that's the one I meant.. self-referencing issues do not help :-)

To keep the discussion in once place, can you copy your comments above, and we can close this one?

**Comment by igrigorik** [comment]
Let's continue in https://github.com/WICG/feature-policy/issues/15.
